### PR TITLE
Test: instrumented test for NoteEditor when text is sent

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
it would be greatly appreciated if you could follow up with a regression test in `androidTest`

_Originally posted by @david-allison in https://github.com/ankidroid/Anki-Android/issues/15566#issuecomment-1950966979_

* https://github.com/ankidroid/Anki-Android/issues/15566#issuecomment-1950966979_

## How Has This Been Tested?
Google emualtor and ran test locally 

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorIntentTest.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+import java.util.concurrent.atomic.AtomicReference
+
+@RunWith(AndroidJUnit4::class)
+class NoteEditorIntentTest {
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorIntentTest.kt:38]: Is there a reason you're not inheriting from `InstrumentedTest`?

**Review by david-allison — CHANGES_REQUESTED** [review]
LGTM, a few nitpicks

Could you change the commit message to:

```
test(note-editor): text shared via ACTION_SEND

Issue 15566
```

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 2]
```diff
@@ -2397,5 +2390,13 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
...
                 .getBoolean(PREF_NOTE_EDITOR_SHOW_TOOLBAR, true)
         }
+
+        @VisibleForTesting
+        @CheckResult
+        fun intentLaunchedWithImage(intent: Intent): Boolean {
+            if (intent.action != Intent.ACTION_SEND && intent.action != Intent.ACTION_VIEW) return false
+            if (ImportUtils.isInvalidViewIntent(intent)) return false
+            return intent.resolveMimeType()?.startsWith("image/") == true
+        }
```
[the remark this answers is not in the document]
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt]: This method is used in the test hence I moved it to companion object

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorIntentTest.kt (hunk 1 above)
Comment (criticalAY) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorIntentTest.kt]: Switched to `InstrumentedTest`

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
LGTM - thank you!

**Pull request merged**